mobile/src/events/Schedule.tsx

```diff
@@ -26,51 +27,38 @@ const ScheduleHomePage: React.FC<{}> = ({ navigation }) => {
 };
 
 const Schedule: React.FC<{}> = () => {
-  const currUser = useSelector((state) => state.auth.user);
-  let currUserId = undefined;
-  if (currUser) {
-    currUserId = currUser.id;
-  }
-
-  // eventsForUser will store the array if events of the current user
   const [eventsForUser, setEventsForUser] = useState([]);
   let today = new Date();
 
-  useEffect(() => {
-    fetch(
-      // add in API
-      `http://localhost:8000/events/${currUserId}`,
-      {
-        method: "GET",
-      }
-    )
-      .then(
-        (res) => {
-          return res.json();
-        },
-        (err) => {
-          console.log(err);
-        }
-      )
-      .then((res) => {
-        if (res) {
-          let newRes = [];
-          res.forEach((e) => {
-            if (new Date(e.start_time).getDate() === today.getDate()) {
-              newRes.push(e);
-            }
-          });
-          // sort events by start_time
-          setEventsForUser(newRes);
-        } else {
-          setEventsForUser([]);
+  React.useEffect(() => {
+    async function fetcher() {
+      try {
+        console.log("hello in try of fetch");
+        const authToken = await SecureStore.getItemAsync("wigo-auth-token");
+        const res = await fetch(
+          'http://localhost:8000/events/currUsersEvents',
+          {
+            headers: {
+              "Content-Type": "application/json;charset=utf-8",
+              "x-auth-token": authToken!,
+            },
+          }
+        );
+        
+        const json = await res.json();
+        const returnedEvents = json.events;
+        console.log("returned events: ", returnedEvents);
+        if (returnedEvents) {
+          setEventsForUser(returnedEvents);
         }
-      })
-      .catch((error) => {
```

Comment:
We don't need a catch block if you're already doing this with the other catch further down. Let's try to use try { ... } catch { ... } and awaits instead of .then and .catch